library/bigip_asm_policy.py

```diff
@@ -0,0 +1,546 @@
+
+    def update(self):
+        policy = self.return_policy()
+        self.have = Parameters(policy.attrs)
+        if not self.should_update():
+            return False
+        if self.client.check_mode:
+            return True
+        if self.update_on_device():
+            return True
+        else:
+            return False
+
+    def should_update(self):
+        result = self._update_changed_options()
+        if result:
+            return True
+        return False
+
+    def update_on_device(self):
+        if self.want.active:
+            if self.is_activated():
+                return False
+            else:
+                if self.client.check_mode:
+                    return True
+                self.activate()
+                return True
+        else:
+            if not self.is_activated():
+                return False
+            if self.client.check_mode:
+                return True
+            self.deactivate()
+            return True
+
+    def install(self):
+        template = self.want.template
+        path = self.want.file
+        if template is None and path is None:
+            self.create_blank()
+            return True
+        if template is None:
+            task = self.import_policy_to_device()
+        if path is None:
+            task = self.create_policy_from_template_on_device()
+        if task:
+            if self.wait_for_task(task):
+                return True
+            else:
+                raise F5ModuleError('Import policy task failed.')
+        return False
+
+    def create_blank(self):
+        self.create_policy_on_device()
+        if self.policy_exists_on_device():
+            return True
+        else:
+            raise F5ModuleError('Failed to create ASM policy: {0}'.format(self.want.name))
+
+    def delete(self):
+        result = self.delete_policy_on_device()
+        return result
+
+    def activate(self):
+        task = self.apply_policy_on_device()
+        if self.wait_for_task(task):
+            return True
+        else:
+            raise F5ModuleError('Apply policy task failed.')
+
+    def deactivate(self):
+        result = self.deactivate_policy_on_device()
+        if result:
+            return True
+        else:
+            raise F5ModuleError('Policy deactivation failed.')
+
+    def wait_for_task(self, task):
+        while True:
+            task.refresh()
+            if task.status in ['COMPLETED', 'FAILURE']:
+                break
+            time.sleep(1)
+        if task.status == 'FAILURE':
+            return False
+        if task.status == 'COMPLETED':
+            return True
+
+    def is_activated(self):
+        result = self.policy_active()
+        if self.client.check_mode:
+            return True
+        return result
+
+    def policy_active(self):
+        policy = self.return_policy()
+        if policy.active is True:
+            return True
+        else:
+            return False
+
+    def return_policy(self):
```

Comment:
this should be named `read_current_from_device` and should look like the method shown here

https://github.com/F5Networks/f5-ansible/blob/devel/library/bigip_device_sshd.py#L256